src/main/java/org/folio/marc4ld/service/ld2marc/resource/Resource2MarcRecordMapperImpl.java

```diff
@@ -117,11 +118,14 @@ private DataField getDataField(Bibframe2MarcFieldRuleApplier b2mRule, Resource r
   private void addInternalIds(Record marcRecord, Resource resource) {
     if (nonNull(resource.getInstanceMetadata())) {
```

Comment:
suggest to refactoring this part
```
    if (nonNull(resource.getInstanceMetadata())) {
      var metaData = resource.getInstanceMetadata();
      if (allNull(metaData.getInventoryId(), metaData.getSrsId())) {
        return;
      }
```java
looks like a separate check
as a suggestion
```
 private void addInternalIds(Record marcRecord, Resource resource) {
    if (nonContainsMetadata) {
return
}
... // other code
```

Reply:
please clarify value of nonContainsMetadata, because if instanceMetadata is not null and srsId and inventoryId are null then bug will be reproducible again.

Reply:
did you mean?
```
if (nonNull(resource.getInstanceMetadata()) 
      || allNull(resource.getInstanceMetadata().getInventoryId(), resource.getInstanceMetadata().getSrsId())) {
      var field999 = marcFactory.newDataField(FIELD_UUID, SPACE, SPACE);
      ofNullable(resource.getInstanceMetadata().getInventoryId()).ifPresent(
        id -> field999.addSubfield(marcFactory.newSubfield(SUBFIELD_INVENTORY_ID, id)));
      ofNullable(resource.getInstanceMetadata().getSrsId()).ifPresent(
        id -> field999.addSubfield(marcFactory.newSubfield(S, id)));
      marcRecord.addVariableField(field999);
    }
```

Reply:
Yeah, but all in "if` looks better to separate method with understanding name

Reply:
fixed